x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/ApiKeySearchQueryBuilder.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.authc.support;
+
+import org.apache.lucene.search.Query;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.PrefixQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.index.query.QueryRewriteContext;
+import org.elasticsearch.index.query.RangeQueryBuilder;
+import org.elasticsearch.index.query.SearchExecutionContext;
+import org.elasticsearch.index.query.TermQueryBuilder;
+import org.elasticsearch.index.query.TermsQueryBuilder;
+import org.elasticsearch.index.query.WildcardQueryBuilder;
+import org.elasticsearch.xpack.core.security.action.GetApiKeyRequest;
+import org.elasticsearch.xpack.core.security.authc.Authentication;
+import org.elasticsearch.xpack.security.authc.ApiKeyService;
+
+import java.io.IOException;
+import java.util.Set;
+
+public class ApiKeySearchQueryBuilder {
+
+    private ApiKeySearchQueryBuilder() {}
+
+    public static BoolQueryBuilder build(GetApiKeyRequest getApiKeyRequest, Authentication authentication) {
+        BoolQueryBuilder finalQuery;
```

Comment:
Why not building an ApiKeyBoolQueryBuilder directly ?

Reply:
Good point. No need to keep both `ApiKeySearchQueryBuidler` and `ApiKeyBoolQueryBuilder`. I consolidated everything into `ApiKeyBoolQueryBuilder`. Thanks!